projects/batfish-client/src/org/batfish/client/Client.java

```diff
@@ -380,86 +372,94 @@ static void validateType(JsonNode value, Variable variable)
       switch (expectedType) {
          case BOOLEAN:
             if (!value.isBoolean()) {
-               throw new BatfishException("");
+               throw new BatfishException(
+                     String.format("Value: %s, is not a valid Boolean " +
+                           "value.", value.toString()));
             }
             break;
          case COMPARATOR:
             if (!(COMPARATORS.contains(value.textValue()))) {
-               throw new BatfishException("");
+               throw new BatfishException(
+                     String.format("Value: %s, is not a valid comparator.",
+                           value.toString()));
             }
             break;
          case INTEGER:
             if (!value.isInt()) {
-               throw new BatfishException("");
+               throw new BatfishException(
+                     String.format("Value: %s, is not a valid Integer.",
+                           value.toString()));
             }
             break;
          case IP:
             if (!value.isTextual()) {
-               throw new BatfishException("");
+               throw new BatfishException(
+                     "IP address was expected to be a String.");
             }
             // TODO: Need to double check isInetAddress()
             if (!InetAddresses.isInetAddress(value.textValue())) {
                throw new BatfishException(
-                     String.format("The value: %s, is not a valid IP address " +
-                                 "for expected type: %s.",
-                                 value.textValue(), expectedType));
+                     String.format("Value: %s, is not a valid IP address.",
+                                 value.textValue()));
             }
             break;
          case IP_WILDCARD:
             if (!value.isTextual()) {
-               throw new BatfishException("");
+               throw new BatfishException(
+                     "IP_Wildcard was expected to be a String.");
```

Comment:
As we discussed, figure out a way to write the variable types that make sense to the user. We can expect the user knows:

* The proper type names used in JSON: that is `expectedType.getName()`.
* English and also terms like Batfish and JSON.

But we cannot expect the user has read this code.

So, we should only use things we know the user knows.

For example:

"A Batfish ipWildcard must be a JSON string."  // plus make sure the actual JSON type plus value is printed somewhere.

or

"An IP wildcard address must be a JSON string." // same caveat.